src/command_modules/azure-cli-extension/azure/cli/command_modules/extension/custom.py

```diff
@@ -139,9 +139,13 @@ def _add_whl_ext(source, ext_sha256=None, pip_extra_index_urls=None):  # pylint:
     extension_path = get_extension_path(extension_name)
     pip_args = ['install', '--target', extension_path, ext_file]
 
+    if pip_proxy:
+        pip_args = pip_args + ['--proxy', pip_proxy]
     if pip_extra_index_urls:
+        index_accum = []
         for extra_index_url in pip_extra_index_urls:
-            pip_args = pip_args + ['--extra-index-url', extra_index_url]
+            index_accum = index_accum + ['--extra-index-url', extra_index_url]
```

Comment:
I am curious to know the difference between the old and the new change. The code behavior looks exactly the same to me.

Reply:
Before this change, for each extra_index_url  the loop was accumulating against the base pip_args including the ['install', '--target', extension_path, ext_file] ... with 2 index values passed in you would have 2x install commands passed in i.e. 

['install', '--target', extension_path, ext_file, '--extra-index-url', index1, 'install', '--target', extension_path, ext_file, '--extra-index-url', index2]

This change uses an empty list for accumulation so you end up with:

['install', '--target', extension_path, ext_file, '--extra-index-url', index1, '--extra-index-url', index2]

Reply:
That being said I don't see documentation supporting multiple --extra-index-url's as this implementation seems to be around.

Reply:
I am not sure why you say "Even with updating pip_extra_index_urls I don't think pip supports multiple --extra-index-url."
I did test adding our extension using the --pip-extra-index-urls option. But, I used only one extra index. There are tests in test_extension_commands.py  that supply two index urls, those urls are not supplying any new packages but at least the code doesn't break with two extra index urls.

This is the actual pip code for extra-index-url option https://github.com/pypa/pip/blob/master/src/pip/_internal/cmdoptions.py#L260

If you have any example that I can run then please let me know.

Reply:
If you run the simple example below, you will see that the pip_args gets the expected value.

pip_args = ['install', '--target', 'extension_path', 'ext_file']
extra_list = ['1', '2', '3', '4']
for item in extra_list:
    pip_args = pip_args + [item]

print(pip_args)

Reply:
Apart from the pip code and my tests that I pointed out, "pip install -h" says that --extra-index-url can be used to specify multiple index URLs.

Reply:
There are 2 things being conflated in this comment.

First is this current code change which resolves having multiple 'install --target ... ...' when having more than one value for --pip-extra-index-urls being passed in. Try passing in 2 indexes with the existing commit with '--debug' appended and you should see logger output right before the pip call.

The second thing is validating/verifying whether pip supports multiple --extra-index-url's. Looking at the [pip documentation](https://pip.pypa.io/en/stable/reference/pip_wheel/#extra-index-url) it is not immediately obvious.

Looking at the pip code you linked looks like it supports it via the 'append' action.

Reply:
@adyada , you are right. My mistake.

Reply:
I run 

az extension add -s C:\Users\adyada\Documents\gitFolder\vienna\src\azureml-cli\src\azure-cli-machinelearning\dist\azure_cli_machinelearning-0.0.97-py2.py3-n
one-any.whl --pip-extra-index-urls https://azuremldownloads.azureedge.net/python-repository/preview https://testpypi.python.org/simple https://pypi.python.org/simple --debug


and get this from debug output, which looks correct to me.

Running: ['C:\\Program Files (x86)\\Python36-32\\python.exe', '-m', 'pip', 'install', '--target', 'C:\\Users\\adyada\\.azure\\cliextensions\\azure-cli-machinelearning', 'C:\\Users\\adyada\\Documents\\gitFolder\\vienna\\src\\azureml-cli\\src\\azure-cli-machinelearning\\dist\\azure_cli_machinelearning-0.0.97-py2.py3-none-any.whl', '--extra-index-url', 'https://azuremldownloads.azureedge.net/python-repository/preview', '--extra-index-url', 'https://testpypi.python.org/simple', '--extra-index-url', 'https://pypi.python.org/simple', '-vv', '--disable-pip-version-check', '--no-cache-dir']


Thanks for digging into this :) --extra-index-url pip argument is confusing. There is also a thread on stack overflow where people were criticizing that it should be --extra-index-urls. But it is an action=append, so --extra-index-url this makes sense.